### DISCUSSION TITLE
Asignación de puntajes y tests

### DISCUSSION THREAD

**Discussion by Maquin01** [IIC2133-PUC/2024-2#113] [Tarea 1] [answered]
He leído varias issues pero no encontré una pregunta exactamente igual a esta:
Mi codigo corre todos los test a excepción del "extreme" de la parte 1 y comienza a fallar en la linea 500.000 (tengo que arreglarlo). Tengo además 2 funciones que son O(n) en vez de O(log(n)) que son enter y leave. 
Como funcionaría exactamente la asignación de puntaje en un caso como este? 
Entiendo que la asignación de enter y leave sería 0 puntos pero, el resto estaría correcto?
Los test para poner la nota son los mismos entregados para probar nuestro código?
Tener este problema en el test "extreme" significa 0 puntos en este test?
Cuantos puntos vale cada test si es que son los mismos que nos dan para probar?
El easy vale menos que el mid y así? o cada uno vale lo mismo?

**Comment by sebacarrascop** [comment, marked as answer]
Hola,

La asignación de puntaje es completamente automática. Si bien se pide que las funciones cumplan con la complejidad esperada, es posible que aún puedas pasar algunos tests incluso sin cumplirla (esperamos que esto no ocurra). Sin embargo, si no alcanzas la complejidad requerida, no obtendrás los puntos correspondientes en el informe por ese aspecto.

Los tests que se usan para evaluar no son exactamente los mismos que los que entregamos para probar su código, pero los tests de evaluación tendrán parámetros similares. Si tu código pasa los tests públicos, es muy probable que también pase los tests privados (los cuales serán utilizados después de la evaluación), sin embargo debes considerar que pueden existir casos bordes que no están siendo evaluados en el test público, pero si podrían ser evaluados en algún privado (esto es ya que los tests son generados a partir de un script).

En cuanto a la cantidad de puntos por test, haremos 3 tests privado por cada test público que se te entregó (cumpliendo que tendrán parámetros similares como mencione anteriormente), y todos valen lo mismo, sin importar si son "easy", "mid", etc. 

Si no cumples con la complejidad en el test "extreme", es probable que obtengas 0 puntos en los tests privados correspondientes a ese test.

Saludos, quedo atento por si te quedaron dudas.

**Reply by Maquin01** [reply]
Muchas gracias por tu respuesta fue muy completa :)
Una duda solo para corroborar:
Si yo tengo, imaginemos, solo una funcion que no cumple con la complejidad requerida, por ejemplo medians. Entonces todos los tests que usen medians tendran 0 puntos? o por cada dificultad hay un test para cada funcion y en los de esa funcion obtendre 0 puntos?
